glacierscript.py

```diff
@@ -372,6 +372,48 @@ def create_unsigned_transaction(source_address, destinations, redeem_script, inp
     return tx_unsigned_hex
 
 
+def find_pubkeys_in_script(decoded_script):
+    """
+    Find the N pubkeys in the provided redeem script
+    Return => List<string> hex-encoded pubkeys
+
+    decoded_script: <dict> output of `decodescript` run on our redeem script
+    """
+    if decoded_script["type"] != "multisig": # sanity check
+        print "ERROR: Supplied redeem script does not seem to be a valid multisig script, Exiting..."
+        sys.exit()
+    # Example "asm": "2 03d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e2 028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b 0315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c1 022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a983 4 OP_CHECKMULTISIG",
+    opcodes = decoded_script["asm"].split()
+    keys = opcodes[1:-2]
+    key_count = int(opcodes[-2])
+    if len(keys) != key_count: # sanity check
+        print "ERROR: Unable to decode supplied redeem script, Exiting..."
+        sys.exit()
+    return keys
+
+def teach_address_to_wallet(source_address, redeem_script, keys):
+    """
+    Teaches the bitcoind wallet about our multisig address, so it can
+    use that knowledge to sign the transaction we're about to create.
+
+    source_address: <string> multisig address
+    redeem_script: <string>
+    keys: List<string> The private keys you wish to sign with
```

Comment:
nit: can we rename this parameter to privKeys or private_keys for clarity?

Reply:
also does "teaching" the wallet about the private keys mean we should take extra care during our tear-down steps in the "Withdraw Protocol"?  I'm not as familiar if there are any cached files on bitcoind that we should be looking out for.

Reply:
I will rename that param.

That's an interesting point. Yes bitcoind will save the privkeys to its wallet.dat. Is that dangerous? I don't think so, since the entire quarantined system is running from RAM disk. Once the system is powered off it's all gone.

But this is a change from the way GlacierScript used to work, and it's possible that I've introduced a new attack vector. Even the possibility is probably good reason to change my approach here. Let me think on this for a bit.